tests/lib/rules/jsx-one-expression-per-line.js

```diff
@@ -0,0 +1,839 @@
+  }, {
+    code: [
+      '<App>',
+      '  <Foo />',
+      '</App>'
+    ].join('\n')
+  }, {
+    code: [
+      '<App>',
+      '  <Foo />',
+      '  <Bar />',
+      '</App>'
+    ].join('\n')
+  }, {
+    code: [
+      '<App>',
+      '  <Foo></Foo>',
+      '</App>'
+    ].join('\n')
+  }, {
+    code: [
+      '<App>',
+      '  foo bar baz  whatever  ',
+      '</App>'
+    ].join('\n')
+  }, {
+    code: [
+      '<App>',
+      '  <Foo>',
+      '  </Foo>',
+      '</App>'
+    ].join('\n')
+  }, {
+    code: [
+      '<App',
+      '  foo="bar"',
+      '>',
+      '<Foo />',
+      '</App>'
+    ].join('\n')
+  }, {
+    code: [
+      '<',
+      'App',
+      '>',
+      '  <',
+      '    Foo',
+      '  />',
+      '</',
+      'App',
+      '>'
+    ].join('\n')
+  }],
+
+  invalid: [{
+    code: '<App>{"foo"}</App>',
+    output: [
+      '<App>',
+      '{"foo"}',
+      '</App>'
+    ].join('\n'),
+    errors: [{message: '`{"foo"}` must be placed on a new line'}],
+    parserOptions: parserOptions
+  }, {
+    code: '<App>foo</App>',
+    output: [
+      '<App>',
+      'foo',
+      '</App>'
+    ].join('\n'),
+    errors: [{message: '`foo` must be placed on a new line'}],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<div>',
+      '  foo {"bar"}',
+      '</div>'
+    ].join('\n'),
+    output: [
+      '<div>',
+      '  foo ',
+      '{\' \'}',
+      '{"bar"}',
+      '</div>'
+    ].join('\n'),
+    errors: [
+      {message: '`{"bar"}` must be placed on a new line'}
+    ],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<div>',
+      '  {"foo"} bar',
+      '</div>'
+    ].join('\n'),
+    output: [
+      '<div>',
+      '  {"foo"}',
+      '{\' \'}',
+      ' bar',
+      '</div>'
+    ].join('\n'),
+    errors: [
+      {message: '` bar` must be placed on a new line'}
+    ],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<App>',
+      '  <Foo /><Bar />',
+      '</App>'
+    ].join('\n'),
+    output: [
+      '<App>',
+      '  <Foo />',
+      '<Bar />',
+      '</App>'
+    ].join('\n'),
+    errors: [{message: '`Bar` must be placed on a new line'}],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<div>',
+      '  <span />foo',
+      '</div>'
+    ].join('\n'),
+    output: [
+      '<div>',
+      '  <span />',
+      'foo',
+      '</div>'
+    ].join('\n'),
+    errors: [{message: '`foo` must be placed on a new line'}],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<div>',
+      '  <span />{"foo"}',
+      '</div>'
+    ].join('\n'),
+    output: [
+      '<div>',
+      '  <span />',
+      '{"foo"}',
+      '</div>'
+    ].join('\n'),
+    errors: [{message: '`{"foo"}` must be placed on a new line'}],
+    parserOptions: parserOptions
+  }, {
+    code: [
+      '<div>',
+      '  {"foo"} { I18n.t(\'baz\') }',
+      '</div>'
+    ].join('\n'),
+    output: [
+      '<div>',
+      '  {"foo"} ',
```

Comment:
this trailing space should be removed, if the space on the next line is being explicitly preserved?

Reply:
React parser already ignores trailing spaces, and `no-trailing-spaces` will chop them off anyway... but let me see if I can get rid of them

Reply:
i agree that no-trailing-spaces could do the cleanup, but it seems easy enough to `.trim()` any newly autofixed lines :-)

Reply:
See latest commit, best I can do right now.